File: 백준/Silver/9461. 파도반 수열/파도반 수열.py
dp = [1, 1, 1, 2, 2]
for i in range(5, 101):
    dp.append(dp[i-1]+dp[i-5])

T = int(input())
for _ in range(T):
    num = int(input())
    print(dp[num-1])

File: 백준/Silver/9461. 파도반 수열/파도반 수열.cc
#include <iostream>
#define MAX 101
using namespace std;

int main()
{
    int N;
    cin >> N;
    
    long long int dp[MAX] = {1, 1, 1, 2, 2};
    for (int i=5; i<MAX; i++)
        dp[i] = dp[i-1] + dp[i-5];
        
    for (int i=0; i<N; i++) {
        int num;
        cin >> num;
        cout << dp[num-1] << "\n";
    }
}